front/pages/api/w/[wId]/vaults/[vId]/data_sources/index.ts

```diff
@@ -0,0 +1,542 @@
+              data_source_error: dustDataSource.error,
+            },
+          });
+        }
+
+        const dataSource = await DataSourceResource.makeNew(
+          {
+            name,
+            description,
+            dustAPIProjectId: dustProject.value.project.project_id.toString(),
+            workspaceId: owner.id,
+            assistantDefaultSelected,
+            editedByUserId: user.id,
+          },
+          vault
+        );
+
+        const dataSourceView =
+          await DataSourceViewResource.createViewInVaultFromDataSourceIncludingAllDocuments(
+            dataSource.vault,
+            dataSource
+          );
+
+        const dataSourceType = await getDataSource(auth, dataSource.name);
+        if (dataSourceType) {
+          void ServerSideTracking.trackDataSourceCreated({
+            user,
+            workspace: owner,
+            dataSource: dataSourceType,
+          });
+        }
+
+        return res.status(201).json({
+          dataSourceView: dataSourceView.toJSON(),
+        });
+      }
+
+      /**
+       * Posting a managed data source
+       * (all providers including webcrawler)
+       */
+      const bodyValidation = PostManagedDataSourceRequestBodySchema.decode(
+        req.body
+      );
+      if (isLeft(bodyValidation)) {
+        const pathError = reporter.formatValidationErrors(bodyValidation.left);
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message: `Invalid request body to post a static data source: ${pathError}`,
+          },
+        });
+      }
+
+      const { connectionId, provider } = bodyValidation.right;
+
+      // Checking that the provider is allowed for the workspace plan
+      const isDataSourceAllowedInPlan = isConnectorProviderAllowedForPlan(
+        plan,
+        provider
+      );
+      if (!isDataSourceAllowedInPlan) {
+        return apiError(req, res, {
+          status_code: 401,
+          api_error: {
+            type: "plan_limit_error",
+            message:
+              "Your plan does not allow you to create managed data sources.",
+          },
+        });
+      }
+
+      // Checking that the vault type is correct for the type of data source being posted:
+      // - System vaults only allow managed data sources (provider is set and not webcrawler)
+      // - Regular vaults only allow folder or website data sources
+      // - Global vaults do not allow data sources to be posted
+      const isFolder = !!provider;
+      const isWebsite =
+        isConnectorProvider(provider) && provider === "webcrawler";
+      const isManaged =
+        isConnectorProvider(provider) && provider !== "webcrawler";
+      if (vault.isGlobal()) {
+        return apiError(req, res, {
+          status_code: 403,
+          api_error: {
+            type: "workspace_auth_error",
+            message: "Cannot administrate global vaults.",
```

Comment:
Why this?

The global vault is a regular vault and you can create folders and websites in it no?

Today only builders can do it and we likely want to keep that for now but this seems to refuse any data source creation?

Reply:
Huh I guess it's because we under "Posting a managed data source" bu then whi isFolder right above?

Reply:
Edited, rule is now: system contains only connected. connected are only in system.